backend/pkg/plugins/plugins.go

```diff
@@ -214,14 +216,42 @@ func pluginBasePathListForDir(pluginDir string, baseURL string) ([]string, error
 	return pluginListURLs, nil
 }
 
+func canSendRefresh(c cache.Cache[interface{}]) bool {
+	value, err := c.Get(context.Background(), PluginCanSendRefreshKey)
+	if err != nil {
+		if errors.Is(err, cache.ErrNotFound) {
+			return false
+		}
+
+		logger.Log(logger.LevelError, map[string]string{"key": PluginCanSendRefreshKey},
+			err, "getting plugin-can-send-refresh key")
+	}
+
+	canSendRefresh, ok := value.(bool)
+	if !ok {
+		logger.Log(logger.LevelInfo, nil, nil, "converting plugin-can-send-refresh key to bool")
+	}
+
+	return canSendRefresh
+}
+
 // HandlePluginEvents handles the plugin events by updating the plugin list
 // and plugin refresh key in the cache.
 func HandlePluginEvents(staticPluginDir, pluginDir string,
 	notify <-chan string, cache cache.Cache[interface{}],
 ) {
 	for range notify {
-		// set the plugin refresh key to true
-		err := cache.Set(context.Background(), PluginRefreshKey, true)
+		// Do not set the refresh signal if we cannot send it. We prevent it here
+		// because we only want to send refresh signals that happen after we are
+		// allowed to send them.
+		var err error
+		if canSendRefresh(cache) {
```

Comment:
err := cache.Set(context.Background(), PluginRefreshKey, canSendRefresh(cache))

Reply:
I didn't do it like this because the comment was more clear in this version of the code. But I will update the code and do this.